src/bluesky/super_state_machine/machines.py

```diff
@@ -0,0 +1,275 @@
+                )
+
+            key = translator.translate(key)
+            cls.context.new_methods[method] = utils.generate_setter(key)
+
+            if from_values:
+                from_values = [translator.translate(k) for k in from_values]
+                for s in cls.context.states_enum:
+                    if s in from_values:
+                        cls.context.new_transitions[s].add(key)
+
+    @classmethod
+    def _unpack_named_transition_tuple(cls, item):
+        try:
+            method, key = item
+            from_values = cls.context["states_enum"]
+        except ValueError:
+            method, key, from_values = item
+            if from_values is None:
+                from_values = []
+            if not isinstance(from_values, list):
+                from_values = [from_values]
+
+        return method, key, from_values
+
+    @classmethod
+    def _add_new_methods(cls):
+        """Add all generated methods to result class."""
+        for name, method in cls.context.new_methods.items():
+            if hasattr(cls.context.new_class, name):
+                raise ValueError(f"Name collision in state machine class - '{name}'.")
+
+            setattr(cls.context.new_class, name, method)
+
+    @classmethod
+    def _set_complete_option(cls):
+        """Check and set complete option."""
+        get_config = cls.context.get_config
+        complete = get_config("complete", None)
+        if complete is None:
+            conditions = [
+                get_config("transitions", False),
+                get_config("named_transitions", False),
+            ]
+            complete = not any(conditions)
+
+        cls.context.new_meta["complete"] = complete
+
+    @classmethod
+    def _set_up_config_getter(cls):
+        meta = getattr(cls.context.new_class, "Meta", DefaultMeta)
+        cls.context.get_config = partial(get_config, meta)
+
+    @classmethod
+    def _set_up_translator(cls):
+        translator = utils.EnumValueTranslator(cls.context["states_enum"])
+        cls.context.new_meta["translator"] = translator
+
+    @classmethod
+    def _calculate_state_name(cls):
+        cls.context.state_name = "state"
+        new_state_name = "_" + cls.context.state_name
+        cls.context.new_meta["state_attribute_name"] = new_state_name
+
+    @classmethod
+    def _complete_meta_for_new_class(cls):
+        cls.context.new_meta["transitions"] = cls.context.new_transitions
+        cls.context.new_meta["config_getter"] = cls.context["get_config"]
+        cls.context.new_class._meta = cls.context["new_meta"]
+
+
+class StateMachine(metaclass=StateMachineMetaclass):
+    """State machine."""
+
+    @property
+    def actual_state(self):
+        """Actual state as `None` or `enum` instance."""
+        attr = self._meta["state_attribute_name"]
+        return getattr(self, attr)
+
+    @property
+    def as_enum(self):
+        """Return actual state as enum."""
+        return self.actual_state
```

Comment:
Moved these methods here from `utils.py` since property decorators outside of class definitions didn't pass linting checks. (This _might_ cause unnoticed name collisions, but within our restricted usage in Bluesky, it is extremely unlikely.)